packages/api/src/hooks/useStaleWalletsAccountsList.ts

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react';
+import useWalletAccountsList from './useWalletAccountsList';
+import useCurrencyConfig from './useCurrencyConfig';
+
+/** A custom hook that gets the list of all wallet accounts for the current user. */
+const useStaleWalletsAccountsList = () => {
+    const { data: walletAccountsList, isLoading } = useWalletAccountsList();
+    const { isLoading: isCurrencyConfigLoading } = useCurrencyConfig();
+
+    const [staleWalletAccountsList, setStaleWalletAccountslist] = useState(walletAccountsList);
+    const [staleIsLoading, setStaleIsLoading] = useState(true);
+
+    useEffect(() => {
+        if (!isLoading && !isCurrencyConfigLoading && walletAccountsList && !staleWalletAccountsList) {
```

Comment:
If we create new wallet will the list be updated? just curious 🤔

Reply:
good point, I would assume that new wallet creation do require page refresh, but let me double check how the stuff works @farhan-nurzi-deriv

Reply:
fixed @farhan-nurzi-deriv though not gonna lie, that case gave me a headache :D 
probably will consider bigger refactors in subsequent PRs to accomodate for it in better way,